### ISSUE TITLE
[Java][feign] snake_case parameter names break feign client

### ISSUE THREAD

**Issue by ePaul** [swagger-api/swagger-codegen#5296]
##### Description

Updating the samples for java/feign in current master and then running the tests of the generated client leads to test failures:

```
Tests in error: 
  testDeleteOrder(io.swagger.client.api.StoreApiTest): status 400 reading StoreApi#getOrderById(Long); content:(..)
  testPlaceOrder(io.swagger.client.api.StoreApiTest): status 400 reading StoreApi#getOrderById(Long); content:(..)
```

##### Swagger-codegen version

Yesterday's master (64c7e23a04d68db9ad), not sure if this applies to any released version.

##### Swagger declaration file content or url

[petstore-with-fake-endpoints-models-for-testing.yaml](https://github.com/swagger-api/swagger-codegen/blob/64c7e23a04d68db9ad27f11295141d8df4b413b9/modules/swagger-codegen/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml)

##### Command line used for generation

    bin/java-petstore-feign.sh

##### Steps to reproduce

```bash
bin/java-petstore-feign.sh
mvn verify -P samples -pl :swagger-petstore-feign
```

##### Related issues

I think this might be another instance of #4898, which was uncovered by the parameter name change in #5267. (I noticed this by the failing tests in #5281.)

##### Suggest a Fix

This is now generated:
```java
  /**
   * Delete purchase order by ID
   * For valid response try integer IDs with value &lt; 1000. Anything above 1000 or nonintegers will generate API errors
    * @param orderId ID of the order that needs to be deleted (required)
   */
  @RequestLine("DELETE /store/order/{order_id}")
  @Headers({
    "Content-Type: application/json",
    "Accept: application/json",
  })
  void deleteOrder(@Param("orderId") String orderId);
```
The `@RequestLine` changed from previously `{orderId}` to now `{order_id}`, while the `@Param` didn't change, and now doesn't match anymore.

**Comment by ePaul** [comment]
I did actually have a look at the `@Param({{paramName}})` in the template when doing the analysis for #4898. For query and header parameters, the sanitized name is actually wanted, because the annotations refer to them, mapping the on-the-wire name to the sanitized names. (That is why I said "don't change this".)

For path parameters this does not happen, we just get the whole path (including the templates with their specification-defined path parameter names) pasted in, as this is already the format wanted by Feign.

I see two approaches here:
* change the path for the `@RequestLine` to have the path parameters with their sanitized names (`{orderId}` in our example).
* change the `@Param` to use `{{baseName}}` instead of `{{paramName}}` for path parameters (so it would become `@Param("order_id")` in our example). This is an ugly case distinction in mustache.

**Comment by wing328** [comment]
I prefer the following:
> change the path for the @RequestLine to have the path parameters with their sanitized names ({orderId} in our example).

I actually applied similar fix to PHP server generator before so I'll file a PR to address this.

cc @bbdouglas

**Comment by wing328** [comment]
Closed via #5301
